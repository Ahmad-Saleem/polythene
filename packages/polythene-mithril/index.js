export * from "polythene-mithril-base";
export * from "polythene-mithril-button";
export * from "polythene-mithril-fab";
export * from "polythene-mithril-icon";
export * from "polythene-mithril-raised-button";
export * from "polythene-mithril-ripple";
export * from "polythene-mithril-shadow";
export * from "polythene-mithril-svg";
