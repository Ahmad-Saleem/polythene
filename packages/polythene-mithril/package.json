{
  "name": "polythene-mithril",
  "version": "1.0.0-rc.1",
  "description": "",
  "main": "dist/polythene-mithril.js",
  "module": "dist/polythene-mithril.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build": "npm run rollup"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-mithril-base": "1.0.0-rc.1",
    "polythene-mithril-button": "1.0.0-rc.1",
    "polythene-mithril-card": "1.0.0-rc.1",
    "polythene-mithril-checkbox": "1.0.0-rc.1",
    "polythene-mithril-dialog": "1.0.0-rc.1",
    "polythene-mithril-dialog-pane": "1.0.0-rc.1",
    "polythene-mithril-fab": "1.0.0-rc.1",
    "polythene-mithril-icon": "1.0.0-rc.1",
    "polythene-mithril-icon-button": "1.0.0-rc.1",
    "polythene-mithril-list": "1.0.0-rc.1",
    "polythene-mithril-list-tile": "1.0.0-rc.1",
    "polythene-mithril-menu": "1.0.0-rc.1",
    "polythene-mithril-radio-button": "1.0.0-rc.1",
    "polythene-mithril-raised-button": "1.0.0-rc.1",
    "polythene-mithril-ripple": "1.0.0-rc.1",
    "polythene-mithril-shadow": "1.0.0-rc.1",
    "polythene-mithril-svg": "1.0.0-rc.1",
    "polythene-mithril-switch": "1.0.0-rc.1",
    "polythene-mithril-tabs": "1.0.0-rc.1",
    "polythene-mithril-textfield": "1.0.0-rc.1",
    "polythene-mithril-toolbar": "1.0.0-rc.1"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://visiblearea.com)",
  "license": "MIT"
}

