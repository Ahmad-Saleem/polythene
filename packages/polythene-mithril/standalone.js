export * from "./index";
import { addLayoutStyles, addFastClick } from "polythene-utilities";
import { addTypography, addRoboto } from "polythene-material-design";

addTypography();
addRoboto();
addLayoutStyles();
addFastClick();
