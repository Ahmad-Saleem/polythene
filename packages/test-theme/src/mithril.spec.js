import { runSnapshots } from "../../tests/scripts/mithril-snapshots";
import { renderer } from "polythene-mithril";
import mithrilTests from "./tests-mithril.js";

runSnapshots({
  tests: mithrilTests,
  renderer
});
