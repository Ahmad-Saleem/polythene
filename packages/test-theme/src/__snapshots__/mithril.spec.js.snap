exports[`test No theme: normal button 1`] = `
"<a class=\"pe-button pe-text-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-ripple\"></div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Unaffected button
  </div>
</div></a>
"
`;

exports[`test Theme with deriving component: button (should be bordered with white background) 1`] = `
"<a class=
\"pe-button pe-text-button pe-button--borders tests-custom-theme-secondary-button\"
tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-ripple\"></div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Bordered button
  </div>
</div></a>
"
`;

exports[`test Theme with style variables: FAB (should be red) 1`] = `
"<a class=
\"pe-fab tests-custom-theme-red-fab tests-custom-theme-red-fab\"
tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple\"></div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-fab__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Theme with style variables: button (should be blue) 1`] = `
"<a class=\"pe-button pe-text-button tests-custom-theme-blue-button\"
tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-ripple\"></div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Blue button
  </div>
</div></a>
"
`;

exports[`test Theme with style variables: button (should be red) 1`] = `
"<a class=\"pe-button pe-text-button tests-custom-theme-red-button\"
tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-ripple\"></div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Red button
  </div>
</div></a>
"
`;

exports[`test Theme with style variables: icon (should be red) 1`] = `
"<div class=\"pe-icon pe-icon--regular tests-custom-theme-red-icon\">
  <div class=\"pe-svg\">
    <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
      <path d=
      \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
      </path>
    </svg>
  </div>
</div>
"
`;

exports[`test Theme with style variables: icon button (should have large padding) 1`] = `
"<a class=
\"pe-button pe-icon-button tests-custom-theme-large-icon-button\"
tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-ripple\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-icon-button__content\">
    <div class=\"pe-icon pe-icon--regular\">
      <div class=\"pe-svg\">
        <svg width=\"24\" height=\"24\" viewbox=\"0 0 24 24\">
          <path d=
          \"M7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7zm1-11h-2v3H8v2h3v3h2v-3h3v-2h-3V9z\">
          </path>
        </svg>
      </div>
    </div>
  </div>
</div></a>
"
`;

exports[`test Theme with style variables: list (should have blue borders) 1`] = `
"<div class=\"pe-list pe-list--borders tests-custom-theme-blue-list\">
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\" style=\"\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\" style=\"\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div>
    </div>
  </div>
  <div class=\"pe-list-tile pe-list-tile--subtitle\">
    <div class=\"pe-list-tile__primary\" style=\"\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Mike Eden
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Watch a game
        </div>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`test Theme with style variables: list tile (should have red titles) 1`] = `
"<div class=\"pe-list\">
  <div class=
  \"pe-list-tile pe-list-tile--subtitle tests-custom-theme-red-list-tile\">
  <div class=\"pe-list-tile__primary\" style=\"\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Jennifer Barker
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Starting post doc
        </div>
      </div>
    </div>
  </div>
  <div class=
  \"pe-list-tile pe-list-tile--subtitle tests-custom-theme-red-list-tile\">
  <div class=\"pe-list-tile__primary\" style=\"\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Ali Connors
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Brunch this weekend?
        </div>
      </div>
    </div>
  </div>
  <div class=
  \"pe-list-tile pe-list-tile--subtitle tests-custom-theme-red-list-tile\">
  <div class=\"pe-list-tile__primary\" style=\"\">
      <div class=\"pe-list-tile__content\">
        <div class=\"pe-list-tile__title\">
          Mike Eden
        </div>
        <div class=\"pe-list-tile__subtitle\">
          Watch a game
        </div>
      </div>
    </div>
  </div>
</div>
"
`;
